src/material-experimental/mdc-button/button.html

```diff
@@ -3,7 +3,13 @@
 <ng-content select=".material-icons:not([iconPositionEnd]), mat-icon:not([iconPositionEnd])">
 </ng-content>
 
-<span class="mdc-button__label"><ng-content></ng-content></span>
+<!--
+  We take advantage of the label element to attach the focus indicator so that we don't
+  have to insert one more element. The indicator can't be directly on the button, because
+  MDC uses ::before for high contrast indication and it can't be on the ripple, because it
+  has a border radius and overflow: hidden.
+-->
+<span class="mdc-button__label mat-mdc-focus-indicator"><ng-content></ng-content></span>
```

Comment:
Putting the indicator class here feels a little hacky, but it's the only place where we can put it without having to add another DOM node. If we're fine with an extra node, I can put it on a separate `div`.

Reply:
yeah lets put it in a new div, we did that for a couple components where there was no other obvious good element to use